src/main/java/me/pheric/pcore/database/Database.java

```diff
@@ -0,0 +1,171 @@
+        dataSource = new HikariDataSource();
+        dataSource.setDriverClassName("com.mysql.jdbc.Driver");
+        dataSource.setJdbcUrl(jdbcUrl);
+        dataSource.setUsername(username);
+        dataSource.setPassword(password);
+        dataSource.setPoolName("Core");
+
+        // Configuration settings for the HikariDataSource
+        dataSource.setMaximumPoolSize(10);
+    }
+
+    /**
+     * Updates a database row with parameters
+     *
+     * @param query  The update statement
+     * @param params The parameters to sanitize in the query
+     * @return Returns -1 if there was an error, 0 if nothing was updated and vice-versa for 1
+     */
+    int update(String query, Object... params) {
+        // Try-resources block to automatically close the connection and prepared statement
+        try (Connection connection = getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            // Loops through the parameters array and sets it on the prepared statement
+            for (int i = 0; i < params.length; i++) {
+                // Set the object in the array to the prepared statement
+                // Which pretty much replaces the quotations
+                // in MySQL queries based on index
+                preparedStatement.setObject((i + 1), params[i]);
+            }
+            // Execute the update and return the value
+            return preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            // Indicates an error
+            return -1;
+        }
+    }
+
+    /**
+     * Find results with a MySQL query
+     *
+     * @param query  The query statement
+     * @param params The parameters to sanitize in the query
+     * @return A result object with all the data stored in it
+     */
+    Result find(String query, Object... params) {
+        // Try-resources block to automatically close the connection and prepared statement
+        try (Connection connection = getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            // Loops through the parameters array and sets it on the prepared statement
+            for (int i = 0; i < params.length; i++) {
+                // Set the object in the array to the prepared statement
+                // Which pretty much replaces the quotations
+                // in MySQL queries based on index
+                preparedStatement.setObject((i + 1), params[i]);
+            }
+
+            // Execute the query and grab the result set
+            ResultSet resultSet = preparedStatement.executeQuery();
+            // Initialize a new result based on
+            Result result = new Result();
+
+            // Check if the result set has anymore data for us
+            if (resultSet.next()) {
+                // Loop through the columns
+                for (int i = 1; i <= resultSet.getMetaData().getColumnCount(); i++) {
```

Comment:
What's going on here?

Reply:
Elaborate?

Reply:
Why are you making a new `Result` instead of returning the resultset for the caller to close themself? I'll need to look into this more later, because I haven't dealt with this recently.

Reply:
Because it's much more difficult to handle result sets than importing the results of it into another class.